### DISCUSSION TITLE
can dafoam do a multi-objective optimization under different Mach no.?

### DISCUSSION THREAD

**Discussion by Irisupupup** [mdolab/dafoam#442] [General]
Hi, I'm wondering if this possible by using DAFoam to implement the following multi-flow-condition constraint like this:
![image](https://github.com/mdolab/dafoam/assets/126235163/86d360ae-811f-4968-9524-d429f3017d0b)
From: 刘超宇, 屈峰, 李杰奇, 白俊强, 刘传振, 白鹏, et al. 涡波一体乘波飞行器宽速域气动优化设计研究. Chinese Journal of Theoretical and Applied Mechanics 2023:70–83.

There are two flow conditions, one in Mach 0.4 and one in Mach 5. It seems we have to solve the adjoint equation for both at the same time and optimize the weighted objective function. I haven't seen such case in the tutorials.
 
Moreover, for super or hypersonic flows like Mach 5, can dafoam solvers solve this problem accurately?

Thanks in advance.

**Comment by friedenhe** [comment]
DAFoam supports multi-point optimization: https://github.com/DAFoam/tutorials/tree/main/NACA0012_Airfoil/multipoint. However, it does not support supersonic flow conditions.

**Reply by FR13ndSDP** [reply]
Hi Frieden, thanks for your wonderful work. To my knowledge, `rhoCentralFoam` is capable of solving supersonic flows, it would be really handy if we can have it as a supersonic solver backend. Do you have any plan to do this?

**Reply by friedenhe** [reply]
@FR13ndSDP Unfortunately, we don't have plans to differentiate rhoCentralFoam in the near future. We are mostly focusing on subsonic flow.
